lib/pure/decimal.nim

```diff
@@ -0,0 +1,1040 @@
+##[
```

Comment:
We should have a performance comparison against established decimal128 libraries, let's followup on this point in https://github.com/timotheecour/Nim/issues/690